### PR TITLE
feat[next]: JAX support for connectivities and premap

### PR DESCRIPTION
**Pull request by havogt**
## Description

Implements [gt4py_knowledge#25](https://github.com/GridTools/gt4py_knowledge/pull/25): `premap` with unstructured connectivities under `jax.jit`.

- `JaxArrayConnectivityField` is registered as a pytree node. The neighbor table is the traced child, so it is a runtime argument of the compiled program instead of an inlined constant. The aux data carries the domain, codomain, skip value and a handle to the same buffer compared by identity; `inverse_image` reads the concrete table through the handle under `jax.ensure_compile_time_eval()`, so domain inference narrows under `jit` exactly as it does eagerly. Retracing happens per table buffer: a new wrapper around the same buffer does not retrace, a new buffer does.
- `inverse_image` answers image ranges that cover all non-skip entries from cached table bounds `(min, max, support)` instead of scanning the table, guarded by the skip value not lying in the range.
- Two pre-existing embedded defects fixed as their own commits:
  - `_gather_premap` let the `-1` skip value wrap around to a real element; harmless for the primal, but under `jax.grad` its cotangent flowed into that element (NaN gradients). Skip indices are now mapped to the field's domain start before the gather. The `skip_value == -1` assertion in `premap` is gone with it.
  - `neighbor_sum` after a narrowing `premap` broadcast the full table against the narrowed field (fails on NumPy too). The reduction now restricts the table to the field's domain and masks with the connectivity's own skip value.
- `test_write_local_field[skip_value_mesh]` (embedded-only) compared against NumPy's wrap-around value at skip positions; it now compares valid entries only.

Out of scope, as in the proposal: `shard_map`/distributed, and a mesh object owning connectivities.

Measured against jax 0.11.0: `PyTreeDef.__hash__` still ignores custom-node aux data, closure constants are still inlined by default, `ensure_compile_time_eval` now works under `lax.scan` (it did not at 0.6.2), and `make_jaxpr(...).consts` no longer lists a closed-over table (it only shows up in the lowered module), so the tests check the StableHLO `@main` signature instead.

## Requirements

- [x] All fixes and/or new features come with corresponding tests.
- [ ] Important design decisions have been documented in the appropriate ADR inside the [docs/development/ADRs/](docs/development/ADRs/) folder.

### CODE REVIEW

File: src/gt4py/next/embedded/nd_array_field.py [hunk 1]
```diff
@@ -596,19 +590,71 @@ def inverse_image(self, image_range: common.UnitRange | common.NamedRange) -> co
 
         return new_domain
 
+    @property
+    def _index_table(self) -> core_defs.NDArrayObject:
+        return self._ndarray
```
Comment (havogt) [inline src/gt4py/next/embedded/nd_array_field.py:597]: why this indirection?

File: src/gt4py/next/embedded/nd_array_field.py [hunk 2]
```diff
@@ -596,19 +590,71 @@ def inverse_image(self, image_range: common.UnitRange | common.NamedRange) -> co
...
+
+    def __setitem__(
+        self,
+        index: common.AnyIndexSpec,
+        value: common.Field | core_defs.NDArrayObject | core_defs.IntegralScalar,
+    ) -> None:
+        super().__setitem__(index, value)
+        self._cache.clear()
+        self.__dict__.pop("_image_bounds", None)
```
Comment (havogt) [inline src/gt4py/next/embedded/nd_array_field.py]: Maybe we should forbid modification of a connectivity?
